core/templates/dev/head/pages/exploration-editor-page/services/learner-answer-details-data.service.ts

```diff
@@ -36,12 +37,17 @@ angular.module('oppia').factory('LearnerAnswerDetailsDataService', [
       '<entity_id>');
 
     var _fetchLearnerAnswerInfoData = function() {
-      var learnerAnswerInfoDataUrl = UrlInterpolationService.interpolateUrl(
-        LEARNER_ANSWER_INFO_DATA_URL, {
-          entity_type: 'exploration',
-          entity_id: _expId});
-
-      return $http.get(learnerAnswerInfoDataUrl);
+      return UserService.getUserInfoAsync().then(function(userInfo) {
+        if (userInfo.isLoggedIn()) {
+          var learnerAnswerInfoDataUrl = UrlInterpolationService.interpolateUrl(
+            LEARNER_ANSWER_INFO_DATA_URL, {
+              entity_type: 'exploration',
+              entity_id: _expId});
+          return $http.get(learnerAnswerInfoDataUrl);
+        } else {
+          return {data: {learner_answer_info_data: []}};
```

Comment:
I have a doubt here - Is this the right way to return the data? I get an errror while running the frontend test that the success handler is not called. If I replace this with `$http.get('random')` and add a `$httpBackend.expect` in test and return the data through it, the test work fine. @DubeySandeep can you please guide me on this? Thanks!

Reply:
That's strange the frontend tests are passing here but they were failing locally for me.

Reply:
@ankita240796, is this issue resolved?

Reply:
No, it isn't. The tests still locally fail when I return the data when the user does not have permission to edit. I return it as follows:

`return $q.resolve({learner_answer_info_data: []})`

The error I get is:

```
Error: Expected spy success to have been called with:
	  [ Object({ learner_answer_info_data: [  ] }) ]
	but it was never called.
```

I just want to confirm once, is there anything wrong with how I am returning the data?

Reply:
The tests fail on circleci too.